src/Relation/Nullary/Recomputable.agda

```diff
@@ -0,0 +1,39 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Recomputable types and their algebra as Harrop formulas
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Relation.Nullary.Recomputable where
+
+open import Data.Product.Base using (_×_; _,_; proj₁; proj₂)
+open import Level using (Level)
+
+private
+  variable
+    a b : Level
+    A : Set a
+    B : Set b
+
+------------------------------------------------------------------------
+-- Definition
+
+Recomputable : (A : Set a) → Set a
+Recomputable A = .A → A
+
+------------------------------------------------------------------------
+-- Constructions
+
+_×-recompute_ : Recomputable A → Recomputable B → Recomputable (A × B)
+(rA ×-recompute rB) p = rA (p .proj₁) , rB (p .proj₂)
+
+_→-recompute_ : (A : Set a) → Recomputable B → Recomputable (A → B)
+(A →-recompute rB) f a = rB (f a)
+
+Π-recompute : (B : A → Set b) → (∀ x → Recomputable (B x)) → Recomputable (∀ x → B x)
+Π-recompute B rB f a = rB a (f a)
+
+∀-recompute : (B : A → Set b) → (∀ {x} → Recomputable (B x)) → Recomputable (∀ {x} → B x)
+∀-recompute B rB f = rB f
```

Comment:
Can we add a proof that sigma is *not* recomputable?
Or is that a meta-theoretical result?

Reply:
Oh... great question (there's even actually something funny about the `_×_` case, in that it requires use of the projections, where I might have thought in the presence of eta that pattern-matching would suffice).

Not sure of the answer: the 'classical' picture was/is that what I've started calling the 'Harrop universe' excludes `∃` and `_⊎_`, towards Harrop's theorem about EP and DP being provable even under hypotheses in CL, provided all the assumptions are Harrop...

Danel Ahman remarked to me that all of this was defining a 'funny' universe of containers, so I suppose it's reasonable to ask what other container(functor)s enjoy the Harrop property, but that wasn't my original motivation, which was to understand the notion of being `Recomputable` as a counterpart to the orthodox realisability notion of 'computationally irrelevant' (as distinct from the mere `.` annotation, or more recent efforts on [Definitional Proof-Irrelevance Without K](https://inria.hal.science/hal-01859964v2/document)...).

Long term, what I want is a tactic which can take suitable (and easy to search for) _classical_ proofs of implications, and turn them into intuitionistically/type-theoretically acceptable ones (hence my question on the Zulip about Aczel/Smith-style slash interpretations, which do I think extend the fragment beyond Harrop, but ... It's All A Long Time Ago), by working with the Harrop metatheorem instantiated to the `Decidable`/`Stable` case.

But these things are (to me, at least) interesting research avenues, and (hence?) for the time being, out of scope for the PR ;-). But if you know/can say more about `Σ`, I'd be very interested!